sonarqube/README.md

```diff
@@ -0,0 +1,188 @@
+# Agent Check: SonarQube
+
+## Overview
+
+This check monitors [SonarQube][1].
+
+## Setup
+
+### Installation
+
+The SonarQube check is included in the [Datadog Agent][2] package.
+No additional installation is needed on your server.
+
+### Configuration
+
+<!-- xxx tabs xxx -->
+<!-- xxx tab "Host" xxx -->
+
+#### Host
+
+To configure this check for an Agent running on a host:
+
+##### Metric collection
+
+1. Edit the `sonarqube.d/conf.yaml` file, in the `conf.d/` folder at the root of your
+   Agent's configuration directory to start collecting your SonarQube data.
+   See the [sample sonarqube.d/conf.yaml][3] for all available configuration options.
+
+   This check has a limit of 350 metrics per JMX instance. The number of returned metrics is indicated in the info page.
+   You can specify the metrics you are interested in by editing the configuration below.
+   To learn how to customize the metrics to collect, visit the [JMX Checks documentation][4] for more detailed instructions.
+   If you need to monitor more metrics, contact [Datadog support][5].
+
+2. [Restart the Agent][6].
+
+##### Log collection
+
+1. Enable SonarQube [logging][7].
+
+2. Collecting logs is disabled by default in the Datadog Agent. Enable it in your `datadog.yaml` file:
+
+   ```yaml
+   logs_enabled: true
+   ```
+
+3. Add the following configuration block to your `sonarqube.d/conf.yaml` file. Change the `path` and `service` parameter values based on your environment. See the [sample sonarqube.d/conf.yaml][3] for all available configuration options.
+
+   ```yaml
+   logs:
+     - type: file
+       path: /opt/sonarqube/logs/access.log
+       source: sonarqube
+       service: <SERVICE>
+     - type: file
+       path: /opt/sonarqube/logs/ce.log
+       source: sonarqube
+       service: <SERVICE>
+       log_processing_rules:
+         - type: multi_line
+           name: log_start_with_date
+           pattern: \d{4}\.\d{2}\.\d{2}
+     - type: file
+       path: /opt/sonarqube/logs/es.log
+       source: sonarqube
+       service: <SERVICE>
+       log_processing_rules:
+         - type: multi_line
+           name: log_start_with_date
+           pattern: \d{4}\.\d{2}\.\d{2}
+     - type: file
+       path: /opt/sonarqube/logs/sonar.log
+       source: sonarqube
+       service: <SERVICE>
+       log_processing_rules:
+         - type: multi_line
+           name: log_start_with_date
+           pattern: \d{4}\.\d{2}\.\d{2}
+     - type: file
+       path: /opt/sonarqube/logs/web.log
+       source: sonarqube
+       service: <SERVICE>
+       log_processing_rules:
+         - type: multi_line
+           name: log_start_with_date
+           pattern: \d{4}\.\d{2}\.\d{2}
```

Comment:
```suggestion
   logs:
     - type: file
       path: /opt/sonarqube/logs/access.log
       source: sonarqube
     - type: file
       path: /opt/sonarqube/logs/ce.log
       source: sonarqube
       log_processing_rules:
         - type: multi_line
           name: log_start_with_date
           pattern: \d{4}\.\d{2}\.\d{2}
     - type: file
       path: /opt/sonarqube/logs/es.log
       source: sonarqube
       log_processing_rules:
         - type: multi_line
           name: log_start_with_date
           pattern: \d{4}\.\d{2}\.\d{2}
     - type: file
       path: /opt/sonarqube/logs/sonar.log
       source: sonarqube
       log_processing_rules:
         - type: multi_line
           name: log_start_with_date
           pattern: \d{4}\.\d{2}\.\d{2}
     - type: file
       path: /opt/sonarqube/logs/web.log
       source: sonarqube
       log_processing_rules:
         - type: multi_line
           name: log_start_with_date
           pattern: \d{4}\.\d{2}\.\d{2}
```

drop service since it's optional